Shuttle-back/src/main/java/com/shuttle/driver/Driver.java

```diff
@@ -1,21 +1,61 @@
 package com.shuttle.driver;
 
+
+import java.sql.Timestamp;
+import java.util.List;
+
+import com.shuttle.location.Location;
+import com.shuttle.note.Note;
+import com.shuttle.security.Role;
 import com.shuttle.user.GenericUser;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
 import jakarta.persistence.Entity;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
 import jakarta.persistence.Table;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
 import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Data;
 import lombok.EqualsAndHashCode;
 import lombok.NoArgsConstructor;
 
 @EqualsAndHashCode(callSuper = true)
 @Data
-@NoArgsConstructor
 @AllArgsConstructor
+@NoArgsConstructor
 @Entity
 @Table(name="driver")
 public class Driver extends GenericUser {
     boolean available;
     Long timeWorkedToday;
     boolean blocked;
+    
+    @OneToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "current_location_id", referencedColumnName = "id")
+    Location currentLocation;
```

Comment:
This shouldn't be here; we already have a Location:

`Driver <---- (1, N) ---- Vehicle ---- (N, 1) ----> Location`